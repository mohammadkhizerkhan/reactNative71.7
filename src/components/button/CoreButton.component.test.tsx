import React from 'react';
import CoreButton from './CoreButton.component';
import {fireEvent, render} from '@testing-library/react-native';

describe('Button Component', () => {
  it('render primary button correctly', () => {
    const onPress = jest.fn();
    const primary = render(
      <CoreButton
        buttonLabel={'primry'}
        buttonType={'primary'}
        buttonColor={'red'}
        textColor={'white'}
        testID="BUTTON"
        onPress={() => onPress()}
      />,
    );
    const button = primary.getAllByTestId('BUTTON-BTN');
    expect(button).toHaveLength(2);
    fireEvent.press(button[0]);
    expect(onPress).toHaveBeenCalledTimes(1);
    expect(primary).toMatchSnapshot();
  });

  it('render primary button correctly', () => {
    const primary = render(
      <CoreButton
        buttonLabel={'primary'}
        buttonType={'outline'}
        testID="BUTTON"
        disabled={true}
      />,
    );
    expect(primary).toMatchSnapshot();
  });

  it('render secondary button correctly', () => {
    const onPress = jest.fn();
    const secondary = render(
      <CoreButton
        buttonLabel="secondary"
        buttonType="secondary"
        buttonColor={'red'}
        textColor={'white'}
        onPress={() => onPress()}
        testID="BUTTON"
      />,
    );
    const button = secondary.getAllByTestId('BUTTON-BTN');
    expect(button).toHaveLength(2);
    fireEvent.press(button[0]);
    expect(onPress).toHaveBeenCalledTimes(1);
    expect(secondary).toMatchSnapshot();
  });

  it('disabled button should work correctly', () => {
    const onPress = jest.fn();
    const disabled = render(
      <CoreButton
        buttonLabel="secondary"
        buttonType="secondary"
        buttonColor={'red'}
        textColor={'white'}
        testID="BUTTON"
        onPress={() => onPress()}
        disabled={true}
      />,
    );

    const button = disabled.getAllByTestId('BUTTON-BTN');
    expect(button).toHaveLength(2);
    expect(button).toBeTruthy();
    fireEvent.press(button[0]);
    expect(onPress).not.toHaveBeenCalled();
    expect(disabled).toMatchSnapshot();
  });
});
