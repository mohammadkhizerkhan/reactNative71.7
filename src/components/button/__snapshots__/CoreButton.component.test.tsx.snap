// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component disabled button should work correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 100,
      "margin": 5,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID="BUTTON-BTN"
>
  <Text
    allowFontScaling={false}
    style={
      [
        [
          {
            "fontSize": 16,
            "lineHeight": 26,
            "textAlign": "center",
          },
        ],
        {
          "fontSize": undefined,
        },
      ]
    }
    testID="BUTTON-BTN"
  >
    secondary
  </Text>
</View>
`;

exports[`Button Component render primary button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 100,
      "margin": 5,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID="BUTTON-BTN"
>
  <Text
    allowFontScaling={false}
    style={
      [
        [
          {
            "fontSize": 16,
            "lineHeight": 26,
            "textAlign": "center",
          },
        ],
        {
          "fontSize": undefined,
        },
      ]
    }
    testID="BUTTON-BTN"
  >
    primry
  </Text>
</View>
`;

exports[`Button Component render primary button correctly 2`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 100,
      "margin": 5,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID="BUTTON-BTN"
>
  <Text
    allowFontScaling={false}
    style={
      [
        [
          {
            "fontSize": 16,
            "lineHeight": 26,
            "textAlign": "center",
          },
        ],
        {
          "fontSize": undefined,
        },
      ]
    }
    testID="BUTTON-BTN"
  >
    primary
  </Text>
</View>
`;

exports[`Button Component render secondary button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 100,
      "margin": 5,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID="BUTTON-BTN"
>
  <Text
    allowFontScaling={false}
    style={
      [
        [
          {
            "fontSize": 16,
            "lineHeight": 26,
            "textAlign": "center",
          },
        ],
        {
          "fontSize": undefined,
        },
      ]
    }
    testID="BUTTON-BTN"
  >
    secondary
  </Text>
</View>
`;
